interfaces/builtin/docker_support.go

```diff
@@ -593,7 +603,16 @@ func (iface *dockerSupportInterface) AppArmorConnectedPlug(spec *apparmor.Specif
 	spec.SetSuppressHomeIx()
 	spec.AddSnippet(dockerSupportConnectedPlugAppArmor)
 	if privileged {
+		// note we don't consider classic vs core here because if we're privileged
+		// then we are unconfined and don't need to worry about apparmor overlayfs
+		// bugs
```

Comment:
'privileged' here != unconfined and corresponds to the the `privileged-containers` interface attribute and privileged only gives `/** wl,`, not 'r' but in `dockerSupportConnectedPlugAppArmorCore` you added 'r'. I think the comment needs updating...

Reply:
Ah yes you're correct I had confused `docker-support` with `lxd-support`. The `'r'` is definitely needed for the `dockerSupportConnectedPlugAppArmorCore` snippet to work with overlayfs, but I had not tested running a privileged container with the overlayfs storage-driver. 
I'm not really sure if it is necessary to add `'r'` to the `dockerSupportPrivilegedAppArmor` snippet, running a privileged container seems to work without the `'r'` in `dockerSupportPrivilegedAppArmor`. What do you think?

Reply:
I adjusted the logic and removed the erroneous comment here.